### PR TITLE
Improve LHEReader IO (and optimize skipping events)

### PR DESCRIPTION
**Pull request by bbockelm**
This branch does two things:
1.  Buffers LHEReader IO along 16MB boundaries (both compressed and uncompressed cases).
2.  Skips parsing of events that will be skipped.

When testing (2), I found there was 1.8x speedup when reading compressed files and 1.9x speedup when reading uncompressed files.  This suggests, when there is efficient IO, we are dominated by XML parsing times (not decompression times).

@davidlange6 @bendavid

### CODE REVIEW

File: GeneratorInterface/LHEInterface/src/XMLUtils.cc [hunk 1]
```diff
@@ -243,67 +245,67 @@ StorageInputStream::~StorageInputStream()
...
+             // Storage layer is supposed to throw exceptions instead of returning errors; just-in-case
+        if (rd < 0)
+        {
+            edm::Exception ex(edm::errors::FileReadError);
+            ex << "Error while reading buffered LHE file";
+            throw ex;
+        }
+        buffLoc_=0;
+        buffTotal_=rd;
+        std::cout << "Total buffer data read: " << buffTotal_ << std::endl;
```
Comment (bendavid) [inline GeneratorInterface/LHEInterface/src/XMLUtils.cc]: This cout should probably be removed.

**Comment by bbockelm** [comment]
Good catch - development cout's removed.

**Comment by bendavid** [comment]
+1

**Comment by bbockelm** [comment]
@bendavid @davidlange6 - should I do another PR for 7_6_X?

**Comment by bendavid** [comment]
Did it already
https://github.com/cms-sw/cmssw/issues/10431

**Comment by bbockelm** [comment]
Thanks!

Is there any final summary of the performance improvements?  I know the real world sometimes differs from my development VM.

I suppose just giving me a heads-up when the affected SUSY LHE gets resubmitted.

**Pull request merged**